pkg/bootstrapper/hostname.go

```diff
@@ -0,0 +1,34 @@
+package bootstrapper
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+)
+
+func getAWSMetadataHostname() (string, error) {
```

Comment:
Do we want to have the AWS specific functions within either a file with AWS in the name or a separate provider specific package?

Reply:
Another package for all providers would be preferred with a separate AWS file.

Reply:
done